core/src/main/java/com/linecorp/armeria/server/docs/DocService.java

```diff
@@ -103,18 +110,17 @@ public final class DocService extends SimpleDecoratingHttpService {
                      spiNamedTypeInfoProviders);
     }
 
+    private static final ExecutorService DEFAULT_EXECUTOR = Executors.newSingleThreadScheduledExecutor(
```

Comment:
> Also how about just creating this executor where it's used because it's mostly used only once when the server is started?

Not sure if this is what you meant, but scheduled the executor to be shut down after file loading has completed 🙏